sixtyfps_runtime/corelib/sharedvector.rs

```diff
@@ -483,6 +480,36 @@ fn invalid_capacity_test() {
     let _: SharedVector<u8> = SharedVector::with_capacity(usize::MAX / 2 - 1000);
 }
 
+#[test]
+fn collect_from_iter_with_no_size_hint() {
+    struct NoSizeHintIter {
+        data: Vec<i32>,
+        i: usize,
+    }
+
+    impl Iterator for NoSizeHintIter {
+        type Item = i32;
```

Comment:
For test, I like using more complex types like String or something else that has a drop so that double drop or wrong initialization would be more visible.

Reply:
I'll cover the capacity issue in a separate test function (via `push`), but I'd like to retain this test function to cover the issue in `from_iter` specifically, when we start out with a capacity that's smaller than the final result. That's why I used this wrapper, to trigger the code path where we copy elements specifically. Is there another type from the standard library that would be better to use instead that does *not* start with the exact capacity?

I was thinking of an iterator with `filter()` or so applied, but I'm not sure that's worth it - I'd find this easier to read. What do you think?

Reply:
I meant using String instead of i32 as the Item, because i32 is Copy and don't need to be drop'ed.

Reply:
Ahh, got it - thanks! Will change.